### PR TITLE
[CAMEL-21361] fabric8 kubernetes-client fails to connect to openshift

### PR DESCRIPTION
**Pull request by tdiesler**
https://issues.apache.org/jira/browse/CAMEL-21361

The root cause was a convoluted TCCL issue. In general, when you work with TCCL, you SHOULD reset it to its original value, like so ...

```
        var tccLoader = Thread.currentThread().getContextClassLoader();
        try {
            ClassLoader cl = createClassLoader();
            Thread.currentThread().setContextClassLoader(cl);

            // do stuff

        } finally {
            Thread.currentThread().setContextClassLoader(tccLoader);
        }
```

### CODE REVIEW

File: dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
         </dependency>
 
         <!-- OpenShift -->
+        <dependency>
```
Comment (davsclaus) [inline dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: Is this only used for testing then set test scope

Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: It's use run `run` as well

Comment (davsclaus) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: where is it needed - it was not there before

Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml:58]: I looked again ... and actually no, the run code uses the k8s API. Let me see if it actually works with scope test ...

Comment (tdiesler) [reply dsl/camel-jbang/camel-jbang-plugin-kubernetes/pom.xml]: done

**Pull request merged**